app/controllers/merchant/orders_controller.rb

```diff
@@ -1,5 +1,21 @@
 class Merchant::OrdersController < Merchant::BaseController
 
+
+
+  def update
+    item = Item.find(params[:item_id])
+    order = Order.find(params[:order_id])
+    item_order = ItemOrder.where(item_id:params[:item_id], order_id:params[:order_id])
+    amount = item.inventory - order.amount_wanted(params[:item_id])
+    item.update_attribute(:inventory, amount)
+    item_order[0].update_attributes(:status => "Fulfilled", quantity: 0)
```

Comment:
Is it necessary to have `item_order[0]`? Isn't item_order an ActiveRecord association, and since you're `.where` with two unique IDs, I imagine you only have 1 item in that AR Association

Reply:
At this moment yes, because where returns an array of objects even if it just one.